text/2603-rust-symbol-name-mangling-v0.md

```diff
@@ -746,6 +747,10 @@ Mangled names conform to the following grammar:
 // We use <path> here, so that we don't have to add a special rule for
 // compression. In practice, only a crate root is expected.
 <instantiating-crate> = <path>
+
+// There are no restrictions on the characters that may be used
+// in the suffix following the `.`.
```

Comment:
Shouldn't this also allow `$` in addition to `.`? I think tools out there are supposed to expect this as well, because of [Itanium](https://itanium-cxx-abi.github.io/cxx-abi/abi.html#mangling-general):

>  Mangled names containing $ or . are reserved for private implementation use. Names produced using such extensions are inherently non-portable and should be given internal linkage where possible. 

Here's an example of someone finding a Rust symbol with a `$` suffix rather than a `.` suffix: https://www.reddit.com/r/rust/comments/t200fl/where_does_in_mangled_name_came_from/

Reply:
Interesting, yes. Seems likely.

Reply:
Yes, I agree that this should be added.